src/main/java/frc/robot/subsystems/BallPath/BallPathImpl.java

```diff
@@ -5,53 +5,49 @@
 import ca.team3161.lib.robot.LifecycleEvent;
 import ca.team3161.lib.robot.subsystem.RepeatingPooledSubsystem;
 
-// import ca.team3161.lib.robot.subsystem.RepeatingIndependentSubsystem;
-// import ca.team3161.lib.robot.subsystem.RepeatingPooledSubsystem;
-
-// import java.util.concurrent.TimeUnit;
-
-// import ca.team3161.lib.robot.LifecycleEvent;
-// import frc.robot.subsystems.BallPath.Intake.Intake;
-import frc.robot.subsystems.BallPath.Intake.IntakeImpl;
-// import frc.robot.subsystems.BallPath.Elevator.Elevator;
-import frc.robot.subsystems.BallPath.Elevator.ElevatorImpl;
-// import frc.robot.subsystems.BallPath.Shooter.Shooter;
-import frc.robot.subsystems.BallPath.Shooter.ShooterImpl;
-
+import frc.robot.subsystems.BallPath.Intake.Intake;
+import frc.robot.subsystems.BallPath.Elevator.Elevator;
+import frc.robot.subsystems.BallPath.Shooter.Shooter;
 import edu.wpi.first.wpilibj.Ultrasonic;
 
 public class BallPathImpl extends RepeatingPooledSubsystem implements BallPath {
 
-    private final IntakeImpl intake;
-    private final ElevatorImpl elevator;
+    private final Intake intake;
+    private final Elevator elevator;
     private final Ultrasonic elevatorSensor;
-    private final ShooterImpl shooter;
+    private final Shooter shooter;
+    private String action;
 
-    public BallPathImpl(IntakeImpl intake, ElevatorImpl elevator, Ultrasonic elevatorSensor, ShooterImpl shooter) {
+    public BallPathImpl(Intake intake, Elevator elevator, Shooter shooter, Ultrasonic elevatorSensor) {
         super(20, TimeUnit.MILLISECONDS);
-        // TODO Auto-generated constructor stub
         this.intake = intake;
         this.elevator = elevator;
         this.elevatorSensor = elevatorSensor;
         this.shooter = shooter;
     }
 
+    @Override
+    public void defineResources(){}
+
+    @Override
+    public void task() throws InterruptedException{
+        if (this.action == "START_INTAKE"){
+            if (this.intake.checkIntake()){
+                this.intake.stop();
+            }
+            if (this.intake.checkColour() && !this.checkIfPrimed()){
+                this.intake.start();
+                Thread.sleep(3000);
```

Comment:
You probably don't want to do a `Thread.sleep` here, because this means that the `task` loop will stop running for 3 whole seconds, meaning the operator won't be able to change the current action request. Or rather, they can change the request, but the subsystem won't start acting upon it for up to 3 seconds.

If you need to include elapsed time as a factor in how long this runs for, you can use a timestamp.

```java
private long intakeStartedTime;

public void task() {
  // other stuff
  // ...
  if (this.intake.checkColour() && !this.checkIfPrimed()){
    long now = System.nanoTime(); // current unix timestamp in nanoseconds
    if (this.intakeStartedTime < 0) {
      this.intakeStartedTime = now;
    }
    this.intake.start();
    if (now > this.intakeStartedTime + TimeUnit.SECONDS.toNanos(3)) {
      this.intake.stop();
      this.intakeStartedTime = -1;
      // we have completed this action and don't want to repeat it all over again, so maybe we do this too:
      this.action = "IDLE";
    }
  }
}
```